### PR TITLE
add : search and address management

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
In this PR, I have added search functionality and address management feature.

### Files to check:
- AddressController.js
- server.js
- modal folder
- data-context modifications only
- dataReducer modifications only

### Files to ignore:
- CSS files and minor refactors

Please review this PR and suggest

###[Live Link](https://dazzle-cerashop-git-feat-search-and-address-mgmt-apurvasawant11.vercel.app/)

### CODE REVIEW

File: src/backend/controllers/AddressController.jsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        error,
+      }
+    );
+  }
+};
+
+/**
+ * This handler handles adding items to user's address.
+ * send POST Request at /api/user/address/:addressId
+ * body contains {action} (whose 'type' can be increment or decrement)
```
Comment (jalajcodes) [inline src/backend/controllers/AddressController.jsx]: body contains action?

File: src/backend/controllers/AddressController.jsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    {
+      address: userAddress,
+    }
+  );
+};
+
+/**
+ * This handler handles adding items to user's address.
+ * send POST Request at /api/user/address
+ * body contains {product}
```
Comment (jalajcodes) [inline src/backend/controllers/AddressController.jsx]: nitpick: correct this comment

**Review by jalajcodes — APPROVED** [review]
Everything looks good. Nice work!

**Review by b0llu — APPROVED** [review]
Great clean code! couldn't  find anything to suggest. Great work in adding address management and coupons

File: src/backend/controllers/AddressController.jsx (hunk 1 above)
Comment (ApurvaSawant11) [reply src/backend/controllers/AddressController.jsx]: Will fix the comments

File: src/backend/controllers/AddressController.jsx (hunk 2 above)
Comment (ApurvaSawant11) [reply src/backend/controllers/AddressController.jsx]: Yes...fixed the comments

**Pull request merged**